src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -0,0 +1,664 @@
+            BytesKmerizer kmerizer = new BytesKmerizer(read.getBases(),KLEN);
+            while ( kmerizer.hasNext() )
+            {
+                Kmer kmer = kmerizer.next().canonical(KLEN);
+                if ( !killSet.contains(kmer) )
+                    kmerList.add(new Tuple2<>(kmer,breakpointId));
+            }
+            return kmerList.iterator();
+        }
+    }
+
+    // Class that acts as a mapper from a stream of reads to a stream of <breakpointId,read> pairs.
+    // It knows which breakpoint(s) a read belongs to (if any) by kmerizing the read, and looking up each Kmer in a
+    // map of Kmers onto breakpoints.
+    private static final class ReadPuller
+            implements Iterator<Tuple2<Long,GATKRead>>, Iterable<Tuple2<Long,GATKRead>>
+    {
+        private final Iterator<GATKRead> inputIterator;
+        private final Map<Kmer,List<Long>> readPullingMap;
+        private Iterator<Tuple2<Long,GATKRead>> outputIterator = Collections.emptyIterator();
+
+        public ReadPuller( Iterator<GATKRead> someInputIterator, Map<Kmer,List<Long>> someReadPullingMap )
+        {
+            inputIterator = someInputIterator;
+            readPullingMap = someReadPullingMap;
+        }
+
+        @Override
+        public boolean hasNext()
+        {
+            while ( !outputIterator.hasNext() )
+            {
+                if ( !inputIterator.hasNext() )
+                    return false;
+                outputIterator = processRead(inputIterator.next());
+            }
+            return true;
+        }
+
+        @Override
+        public Tuple2<Long,GATKRead> next()
+        {
+            if ( !hasNext() )
+                throw new NoSuchElementException("ClusterKmerizer is exhausted.");
+            return outputIterator.next();
+        }
+
+        @Override
+        public Iterator<Tuple2<Long,GATKRead>> iterator() { return this; }
+
+        private Iterator<Tuple2<Long,GATKRead>> processRead( GATKRead read )
+        {
+            Set<Long> breakpointSet = new HashSet<>();
+            BytesKmerizer kmerizer = new BytesKmerizer(read.getBases(),KLEN);
+            while ( kmerizer.hasNext() )
+            {
+                Kmer kmer = kmerizer.next().canonical(KLEN);
+                List<Long> breakpoints = readPullingMap.get(kmer);
+                if ( breakpoints != null )
+                    breakpointSet.addAll(breakpoints);
+            }
+
+            if ( breakpointSet.isEmpty() )
+                return Collections.emptyIterator();
+
+            List<Tuple2<Long,GATKRead>> readList = new ArrayList<>(breakpointSet.size());
+            for ( Long breakpointId : breakpointSet )
+                readList.add(new Tuple2<>(breakpointId,read));
+
+            return readList.iterator();
+        }
+    }
+/*
+    // dead code to reorder a nearly coordinate-ordered stream of reads into their original coordinate order
```

Comment:
I'm not sure why you want to check this in, but if you do, please open an issue to remove it before you merge it in.